core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -162,7 +179,7 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   enum { rank = static_cast<int>(iteration_pattern::rank) };
 
   using index_type       = typename traits::index_type;
-  using array_index_type = long;
+  using array_index_type = std::int64_t;
   using point_type = Kokkos::Array<array_index_type, rank>;  // was index_type
```

Comment:
You may ask why didn't I change this to
```suggestion
  using point_type = Kokkos::Array<index_type, rank>;  // was index_type
```
The answer is because it would potentially break user code :/